scintellometry/io/arochime.py

```diff
@@ -166,6 +168,203 @@ def open(self, number=0):
+                                      dtype='cu4bit,cu4bit', comm=comm)
+        self.time0 = self.fh_raw.time0
+        self.samplerate = self.fh_raw.samplerate
+        self.dtsample = (1 / self.samplerate).to(u.s)
+        self.npol = self.fh_raw.npol
+        self.fedge = self.fh_raw.fedge
+        self.fedge_at_top = self.fh_raw.fedge_at_top
+        super(AROCHIMEInvPFB, self).__init__(raw_files, blocksize, dtype,
+                                             nchan=1, comm=comm)
+        # PFB information
+        self.nblock = 2048
+        self.h = pfb.sinc_hamming(4, self.nblock).reshape(4, -1)
+        self.fh = None
+
+        # S/N for use in the Wiener Filter
+        # Assume 8 bits are set to have noise at 3 bits, so 1.5 bits for FT.
+        # samples off by uniform distribution of [-0.5, 0.5] ->
+        # equivalent to adding noise with std=0.2887
+        prec = (1 << 3) ** 0.5
+        self.sn = prec / 0.2887
+        print("Opened InvPFB reader, S/N={0}".format(self.sn))
+
+    def open(self, number=0):
+        self.fh_raw.open(number)
+
+    def close(self):
+        self.fh_raw.close()
+
+    def _seek(self, offset):
+        if offset % self.recordsize != 0:
+            raise ValueError("Cannot offset to non-integer number of records")
+
+        self.offset = (offset // self.fh_raw.recordsize *
+                       self.fh_raw.recordsize)
+
+    def seek_record_read(self, offset, size):
+        if offset % self.recordsize != 0 or size % self.recordsize != 0:
+            raise ValueError("size and offset must be an integer number of records")
+
+        raw = self.fh_raw.seek_record_read(offset, size)
+
+        if self.npol == 2:
+            raw = raw.view(raw.dtype.fields.values()[0][0])
+
+        raw = raw.reshape(-1, self.fh_raw.nchan, self.npol)
+
+        nyq_pad = np.zeros((raw.shape[0], 1, self.npol), dtype=raw.dtype)
+        raw = np.concatenate((raw, nyq_pad), axis=1)
+        # Get pseudo-timestream
+        print('Getting pseudo timestream')
+        pd = np.fft.irfft(raw, axis=1)
+        # Set up for deconvolution
+        print('Setting up for deconvolution')
+        fpd = np.fft.rfft(pd, axis=0)
+        del pd
+        if self.fh is None or self.fh.shape[0] != fpd.shape[0]:
+            print('Getting FT of PFB')
+            lh = np.zeros((raw.shape[0], self.h.shape[1]))
+            lh[:self.h.shape[0]] = self.h
+            self.fh = np.fft.rfft(lh, axis=0).conj()
+            del lh
+        # FT of Wiener deconvolution kernel
+        fg = self.fh.conj() / (np.abs(self.fh)**2 + (1/self.sn)**2)
+        # Deconvolve and get deconvolved timestream
+        print('Wiener deconvolving')
+        rd = np.fft.irfft(fpd * fg[..., np.newaxis],
+                          axis=0).reshape(-1, self.npol)
+        # select actual part requested
+        print('Selecting and returning')
+        self.offset = offset + size
+        # view as a record array
+        return rd.astype('f4').view('f4,f4')
```

Comment:
With the change in `fold.py` suggested above, you can also remove the `view('f4,f4')` here (but I'd leave the `astype`)